File: backend/app/schemas/paper.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class PaperResponse(BaseModel):
    id: int
    project_id: int
    title: str
    filename: str
    file_path: str
    file_size: int
    uploaded_at: datetime

    summary: str | None = None
    page_count: int | None = None
    word_count: int | None = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/paper.ts
export interface Paper {
  id: number;
  project_id: number;

  title: string;
  filename: string;

  file_path: string;

  file_size: number;

  uploaded_at: string;

  page_count: number | null;

  word_count: number | null;

  summary: string | null;
}